tasks/firewalld.yml

```diff
@@ -23,3 +23,27 @@
     state: present
     use: "{{ (__firewall_is_ostree | d(false)) |
       ternary('ansible.posix.rhel_rpm_ostree', omit) }}"
+  register: firewall_package_result
+
+- name: Handle reboot for transactional update systems
+  when:
+    - system_is_transactional | d(false)
```

Comment:
Define those variables in `defaults/main.yml`, set both to `false there. Then, the `| d` filter is not needed.

Reply:
When setting the reboot variable in the defaults, the fail task is bypassed as it looks for 'is undefined'.
__
This variable is used to handle reboots required by transactional updates. If a transactional update requires a reboot, the role will proceed with the reboot if transactional_update_reboot_ok is set to true. If set to false, the role will notify the user that a reboot is required, allowing for custom handling of the reboot requirement. If this variable is not set, the role will fail to ensure the reboot requirement is not overlooked.

Reply:
We have a pretty strong preference for requiring all variables that are part of the public API of the role to be specified in defaults/main.yml with the default value.  In this case, the default value can be `null`. Then the test will be

```yaml
  when:
    - system_is_transactional is not none
    - system_is_transactional | d(false)
```

or if you prefer a one-liner:

```yaml
when: (system_is_transactional is none) | ternary(false, system_is_transactional)
```

We use this convention in many, many places in system roles.